qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,213 @@
+                                     sync, speed, mode, bitmap_name)
+        if not self.get_status():
+            raise error.TestFail("%s backup job not found" % sync)
+
+    def transaction_add(self, args_list, type, data):
+        """
+        Generate a args list for a transaction.
+        :param args_list: the arg_list param that will be modified
+        :param type: type of the command
+        :param data: data of the command
+        """
+        args = {"type": type, "data": data}
+        args_list.append(args)
+
+    def backup_check(self, compare_image=None):
+        """
+        Check and compare the backup images with qemu-img
+        :param compare_image: the image that need to be compared
+        """
+        data_dir = self.data_dir
+        for image in self.image_chain:
+            params = self.params.object_params(image)
+            qemu_image = qemu_storage.QemuImg(params, data_dir, image)
+            qemu_image.check_image(params, data_dir)
+        params = self.params.object_params(self.source_image)
+        if compare_image:
+            compare_image = utils_misc.get_path(data_dir, compare_image)
+            source_image = qemu_storage.storage.get_image_filename(params, data_dir)
+            qemu_image.compare_images(compare_image, source_image)
+
+    def reopen(self):
+        """
+        Closing the vm and reboot it with the backup image.
+        """
+        vm = self.vm
+        vm.destroy()
+        image_chain = self.image_chain
+        image_name = self.params.get("image_name_%s" %
+                                     image_chain[len(image_chain) - 1])
+        self.params["image_name_%s" % self.source_image] = image_name
+        vm.create(params=self.params)
+        self.vm = vm
+        session = vm.wait_for_login(
+            timeout=int(self.params.get("login_timeout", 360)))
+        session.close()
+
+    def before_full_backup(self):
+        """
+        Run steps before create full backup.
+        """
+        return self.do_steps("before_full_backup")
+
+    def before_incremental(self):
+        """
+        Run steps before create incremental backup.
+        """
+        return self.do_steps("before_incremental")
+
+    def after_full_backup(self):
+        """
+        Run steps after create full backup.
+        """
+        self.do_steps("after_full_backup")
+        time.sleep(120)
+
+    def after_incremental(self):
+        """
+        Run steps after create incremental backup.
+        """
+        return self.do_steps("after_incremental")
+
+    def verify_job_info(self):
+        """
+        Check block job settings correctness.
+        """
+        params = self.params
+        device_id = self.get_device()
+        info = self.vm.monitor.info_block().get(device_id)
+        check_params = params.get("check_params")
+        for check_param in check_params:
+            if check_param in params:
```

Comment:
check_params = filter(lambda x: x in params, check_params) or 
check_params = [ _ for _ in check_params if _ in params]